lib/spack/spack/tag.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+"""Classes and functions to manage package tags"""
+import collections
+import copy
+import sys
+
+if sys.version_info >= (3, 5):
+    from collections.abc import Mapping  # novm
+else:
+    from collections import Mapping
+
+import spack.error
+import spack.util.spack_json as sjson
+
+
+def get_installed_packages():
```

Comment:
I'd be fine with this if it is an underscore function:
```suggestion
def _get_installed_packages():
```
If it is part of the public API, it should probably be reworked and moved into `spack.environment` or `spack.store`?. Rationale for this is that the function does not deal with tags at all, just with environments and installed specs.